lxd/tokens.go

```diff
@@ -28,7 +38,20 @@ func autoRemoveExpiredTokens(ctx context.Context, s *state.State) {
 		}
 	}
 
-	if len(expiredTokenOps) == 0 {
+	leaderInfo, err := s.LeaderInfo()
+	if err != nil {
+		logger.Warn("Failed to get database leader details", logger.Ctx{"err": err})
+	}
+
+	var expiredPendingTLSIdentities []cluster.Identity
+	if leaderInfo != nil && leaderInfo.Leader {
+		expiredPendingTLSIdentities, err = getExpiredPendingIdentities(ctx, s)
+		if err != nil {
+			logger.Warn("Failed to retrieve expired pending TLS identities during removal of expired tokens task", logger.Ctx{"err": err})
```

Comment:
why do we continue here?

Reply:
So that any local certificate add/cluster join token operations are still pruned. I've added a comment.